### PR TITLE
Add posts

### PR DESCRIPTION
**Pull request by stavby**
dlugin is a hamor

### CODE REVIEW

File: controllers/posts.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            return;
+        }
+
+        res.status(httpStatus.OK).send('Updated successfully');
+    } catch (error) {
+        next(error);
+    }
+};
+
+export const postsErrorHandler: ErrorRequestHandler = (error, req, res, next) => {
```
Comment (yarondlugin) [inline controllers/posts.ts]: rename to errorHandler

File: controllers/posts.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    console.error(`An error occured in posts router at ${req.method} ${req.url} - `, (error as Error).message);
```
Comment (yarondlugin) [inline controllers/posts.ts]: you can give error the Error type (in the function definition) instead of casting

File: controllers/posts.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import { ErrorRequestHandler, NextFunction, Request, Response } from 'express';
+import httpStatus from 'http-status';
+import { postModel } from '../models/posts';
+import { isValidObjectId } from 'mongoose';
```
Comment (yarondlugin) [inline controllers/posts.ts:3]: external imports before local imports

File: controllers/posts.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+export const createPost = async (request: Request, response: Response, next: NextFunction) => {
+    try {
```
Comment (yarondlugin) [inline controllers/posts.ts]: you don't need try here, move it to before the postmodel.create function call

File: controllers/posts.ts (hunk 3 above)
Comment (yarondlugin) [reply controllers/posts.ts:3]: also in the other functions

File: controllers/posts.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+        next(error);
+    }
+};
+
+export const getAllPosts = async (request: Request<{}, {}, {}, { sender: string }>, response: Response, next: NextFunction) => {
+    try {
+        const { sender } = request.query;
+
+        const posts = !!sender ? await postModel.find({ sender }) : await postModel.find();
```
Comment (yarondlugin) [inline controllers/posts.ts]: maybe `find({ sender: sender ?? null })` ?

File: controllers/posts.ts [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+            return;
+        }
+
+        const post = await postModel.findById(postId);
+        if (!post) {
+            response.status(httpStatus.NOT_FOUND).send(`Post with id ${postId} not found`);
+            return;
+        }
+
+        response.send(post);
```
Comment (yarondlugin) [inline controllers/posts.ts]: missing request status

File: controllers/posts.ts (hunk 5 above)
Comment (stavby) [reply controllers/posts.ts]: Ended up with
`const posts = await postModel.find(!!sender ? { sender } : {});`

**Comment by yarondlugin** [comment]
approved

**Comment by stavby** [comment]
shut up

**Pull request merged**